components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.common/src/main/java/org/wso2/carbon/identity/api/server/application/management/common/ApplicationManagementConstants.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.identity.api.server.application.management.common;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.ResourceBundle;
+import javax.ws.rs.core.Response;
+
+/**
+ * Holds constants related to Application Management components.
+ */
+public class ApplicationManagementConstants {
+
+    private ApplicationManagementConstants() {
+
+    }
+
+    private static final String APPLICATION_MANAGEMENT_PREFIX = "APP-";
+    public static final String APPLICATION_MANAGEMENT_PATH_COMPONENT = "/applications";
+
+    /**
+     * Enums for error messages.
+     */
+    public enum ErrorMessage {
+
+        // Client errors.
+        ERROR_CODE_INVALID_INPUT("50001",
+                Response.Status.BAD_REQUEST,
+                "Invalid input.",
+                "One of the given inputs is invalid."),
+        ERROR_CODE_UNSUPPORTED_FILTER_ATTRIBUTE("50002",
+                Response.Status.BAD_REQUEST,
+                "Filtering using the attempted attribute is not supported.",
+                "Filtering cannot be done with the '%s' attribute. " +
+                        "Filtering is only supported with the 'name' attribute."),
+        ERROR_CODE_INVALID_FILTER_FORMAT("50003",
+                Response.Status.BAD_REQUEST,
+                "Invalid format user for filtering.",
+                "Filter needs to be in the format <attribute>+<operation>+<value>. Eg: name+eq+john"),
+        ERROR_CODE_UNSUPPORTED_FILTER_OPERATION("55004",
```

Comment:
Why this starts with 55 while other are not

Reply:
Shouldn't the error code be not implemented or something denoting unsupported operations rather than identifying as a bad request

Reply:
> Why this starts with 55 while other are not

We have decided on a convention for Server APIs in architecture@wso2.org,

[Architecture] WSO2 Identity Server REST API Error Response Standardization

Reply:
> Shouldn't the error code be not implemented or something denoting unsupported operations rather than identifying as a bad request

I think we should change the error to say Invalid Filter Operation. Because in this case the user has attempted an invalid filter operation.
Will that make sense?

Reply:
+1